src/scheduler/MpiWorld.cpp

```diff
@@ -379,37 +379,32 @@ std::string MpiWorld::getHostForRank(int rank)
     return host;
 }
 
-std::vector<int> MpiWorld::getRanksForHost(const std::string& host)
+// Group the ranks to host vector by host
+void MpiWorld::setRanksForHost()
 {
-    assert(rankHosts.size() == size);
+    assert(hostForRank.size() == size);
 
-    std::vector<int> ranksForHost;
-    for (int i = 0; i < rankHosts.size(); i++) {
-        if (rankHosts.at(i) == host) {
-            ranksForHost.push_back(i);
-        }
+    for (int rank = 0; rank < hostForRank.size(); rank++) {
+        std::string host = hostForRank.at(rank);
```

Comment:
Wrote this in two lines for the sake of clarity, reading `hostForRank` and `ranksForHost` on the same line can be a bit of a tong-twister. Happy to merge to one though.